plugins/modules/mysql_info.py

```diff
@@ -122,6 +125,35 @@
     - databases
     exclude_fields: db_size
     return_empty_dbs: true
+
+- name: Clone users on another server - Step 1
+  delegate_to: server_source
+  community.mysql.mysql_info:
+    filter:
+      - users_privs
+  register: result
+
+# Don't work with sha256_password and cache_sha2_password
+- name: Clone users on another server - Step 2
+  community.mysql.mysql_user:
+    name: "{{ item.name }}"
+    host: "{{ item.host }}"
+    plugin: "{{ item.plugin | default(omit) }}"
+    plugin_auth_string: "{{ item.plugin_auth_string | default(omit) }}"
+    plugin_hash_string: "{{ item.plugin_hash_string | default(omit) }}"
+    tls_require: "{{ item.tls_require | default(omit) }}"
+    priv: "{{ item.priv | default(omit) }}"
+    resource_limits: "{{ item.resource_limits | default(omit) }}"
+    column_case_sensitive: true
+    state: present
+  loop: "{{ result.users_privs }}"
+  loop_control:
+    label: "{{ item.name }}@{{ item.host }}"
+  when:
+    - item.name != 'root'  # In case you don't want to import admin accounts
+    - item.name != 'mariadb.sys'
+    - item.name != 'mysql'
+
```

Comment:
```suggestion
- name: Clone users from one server to another
  block:
  - name: Fetch information from a source server
    delegate_to: server_source
    community.mysql.mysql_info:
      filter:
        - users_privs
    register: result

  # Don't work with sha256_password and cache_sha2_password
  - name: Clone users fetched in a previous task to a target server
    community.mysql.mysql_user:
      name: "{{ item.name }}"
      host: "{{ item.host }}"
      plugin: "{{ item.plugin | default(omit) }}"
      plugin_auth_string: "{{ item.plugin_auth_string | default(omit) }}"
      plugin_hash_string: "{{ item.plugin_hash_string | default(omit) }}"
      tls_require: "{{ item.tls_require | default(omit) }}"
      priv: "{{ item.priv | default(omit) }}"
      resource_limits: "{{ item.resource_limits | default(omit) }}"
      column_case_sensitive: true
      state: present
    loop: "{{ result.users_privs }}"
    loop_control:
      label: "{{ item.name }}@{{ item.host }}"
    when:
      - item.name != 'root'  # In case you don't want to import admin accounts
      - item.name != 'mariadb.sys'
      - item.name != 'mysql'
```
How about using a block to separate this tasks from the rest?